src/test/java/com/develhope/spring/deals/controllers/OrderControllerTest.java

```diff
@@ -0,0 +1,149 @@
+package com.develhope.spring.deals.controllers;
+
+import com.develhope.spring.deals.responseStatus.OrderNotFoundException;
+import com.develhope.spring.deals.services.OrderService;
+import org.json.JSONObject;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+
+import java.util.NoSuchElementException;
+
+import static org.mockito.Mockito.*;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+@DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
+public class OrderControllerTest {

```

Comment:
questi test sono alquanto confusi. Cè un misto di mockito e mvc, when mi pare sia un metodo solo di mockito che quindi non dovresti usare in un test con mock mvc